streams/common/src/main/java/com/hortonworks/streamline/streams/common/event/correlation/GroupedCorrelationEvents.java

```diff
@@ -89,4 +97,48 @@ public String getComponentName() {
             return outputEventIds;
         }
     }
+
+    public static class SortedComponentGroupedEvents {
+        private String componentName;
+        private boolean containingSelectedEvent;
+        private List<String> inputEventIds;
+        private List<String> outputEventIds;
+
+        public SortedComponentGroupedEvents(ComponentGroupedEvents groupedEvents, String selectedEventId) {
+            this.componentName = groupedEvents.getComponentName();
+
+            this.inputEventIds = new ArrayList<>(groupedEvents.getInputEventIds());
+            this.outputEventIds = new ArrayList<>(groupedEvents.getOutputEventIds());
+
+            if (this.outputEventIds.contains(selectedEventId)) {
+                // this component contains selected event Id
+                this.containingSelectedEvent = true;
+
+                this.outputEventIds.remove(selectedEventId);
```

Comment:
nit: can be LinkedList since we are adding and removing entries?

Reply:
Element count in array should be super small enough so that's trivial, but the change is also trivial I'll apply it.

Reply:
Addressed.